CONTRIBUTING.rst

```diff
@@ -328,6 +334,45 @@ detailing the test coverage::
   networkx/algorithms/approximation/clique.py         42      1     18      1    97%
   ...
 
+Adding tests
+------------
+
+If you're **new to testing**, see existing test files for examples of things to do.
+**Don't let the tests keep you from submitting your contribution!**
+If you're not sure how to do this or are having trouble, submit your pull request
+anyway.
+We will help you create the tests and sort out any kind of problem during code review.
+
+General guidelines for writing good tests:
+
+- doctests always assume ``import networkx as nx`` so don't add that
+- avoid test classes when possible
+- use the following decorators (``@pytest.mark.parametrize``, ``@pytest.mark.skipif``, ``@pytest.mark.xfail``, ``@pytest.mark.slow``)
```

Comment:
Of these decorators, I would maybe only call out `parametrize`. I don't know if you necessarily want to *encourage* test authors to use `skipif` or `xfail`, unless they run into a special case where they're warranted (e.g. a test won't work on Windows). I don't think these come up too often in the test suite, so maybe they don't need to be emphasized. Similarly with `slow` - I would say that the emphasis should be "write good tests, regardless of how long they take to run" and then `slow` can be added afterwards if a test has a long run-time.

Reply:
I moved this to the reference guide and I also edited based on your comments.